flumeDevice

```diff
@@ -383,25 +399,58 @@ def queryDevice()
         sendEvent(name: "usageLastMonth", value: (resp.data[0]?.lastMonth && (resp.data[0]?.lastMonth?.size() > 0)) ? resp.data[0].lastMonth.getAt(resp.data[0].lastMonth.size() - 1)?.value?.toFloat().round(2) : "n/a")
 
-        lastFlowTS = getLastPollTS()
+        lookbackFlow = (resp.data[0]?.flowLookback && (resp.data[0]?.flowLookback?.size() > 0) && resp.data[0].flowLookback.getAt(resp.data[0].flowLookback.size() - 1)?.value?.toFloat() > 0) ? true : false
+        refreshFlow = (resp.data[0]?.flowRefresh && (resp.data[0]?.flowRefresh?.size() > 0) && resp.data[0].flowRefresh.getAt(resp.data[0].flowRefresh.size() - 1)?.value?.toFloat() > 0) ? true : false
+
         setLastPollTS()
         currentFlowTS = getLastPollTS()
+        flowDetectedTS = getFlowDetectedTS()
+        flowStatus = device.currentValue("flowStatus")
 
-        lookbackFlow = (resp.data[0]?.alertLookback && (resp.data[0]?.alertLookback?.size() > 0) && resp.data[0].alertLookback.getAt(resp.data[0].alertLookback.size() - 1)?.value?.toFloat() > 0) ? true : false
-        pollWindow = (currentFlowTS - lastFlowTS)/60000
-        if(pollWindow <= flowAlarmLookback && lookbackFlow)
+        if (refreshFlow || lookbackFlow)
         {
-            continuousFlowReading = device.currentValue("continuousFlowReadings") == null ? 1 : device.currentValue("continuousFlowReadings") + 1
-            sendEvent(name: "continuousFlowReadings", value: continuousFlowReading)
+            // [cgmckeever] future us, this `flowDetectedTS == 0` is an edge case
+            //              where water is flowing and preferences are saved.
+            //              `flowDetectedTS` state variable gets purged,
+            //              making runTime very large, and usually triggers an alert
+            if (flowStatus == "stopped" || flowDetectedTS == 0 )
+            {
+                setFlowDetectedTS()
+                flowDetectedTS = getFlowDetectedTS()
+                logDebug(">>>>>>>>>>>>> startFlowWatch: ${flowDetectedTS}")
+            }
 
-            if (flowAlarmTrigger > 0 && continuousFlowReading >= flowAlarmTrigger)
+            if (refreshFlow)
+            {
+                status = "running"
+            } else
             {
-                sendEvent(name: "water", value: "wet")
+                status = "monitoring"
             }
-        }
-        else
+
+            if (flowStatus != status)
+            {
+                sendEvent(name: "flowStatus", value: status)
+            }
+
+            runTime = Math.round((currentFlowTS - flowDetectedTS)/600)/100
+            sendEvent(name: "flowDurationDetail", value: status + " for " + runTime.toString() + " min")
+            sendEvent(name: "flowDurationMin", value: runTime)
+
+            if (flowAlarmTriggerTime > 0 && runTime >= flowAlarmTriggerTime)
```

Comment:
@tomwpublic based on the latest thread. I had a 💡 .. what if we just get rid of the driver controlling an alert, and just let people make their own rules?

- 1 less preference var `flowAlarmTriggerTime`
- Less blame that the driver caused a false alarm since others make their own rules based on the run time?
```
sendEvent(name: "flowDurationMin", value: runTime)
```
- It'll be 4 less lines of code ;)

Reply:
I like that idea.  I was going to spend some time trying to reduce the likelihood of "wet" based on the analysis, since I the guy who was running sirens and closing his main valve made me worry about it being too aggressive.

I think your suggestion will be cleaner -- if we reserve the "wet" status for Flume leak alerts (the current behavior) and use the attributes you created for people to build rules logic around.